frontend/packages/dev-console/src/components/taskruns/TaskRunDetailsErrorLog.tsx

```diff
@@ -0,0 +1,35 @@
+import * as React from 'react';
+import { getLogSnippet } from '../pipelineruns/logs/pipelineRunLogSnippet';
+import LogSnippetBlock from '../pipelineruns/logs/LogSnippetBlock';
+import { TaskRunKind } from '../../utils/pipeline-augment';
+
+type TaskStatusLogProps = {
+  taskRun: TaskRunKind;
+};
+
+const TaskRunDetailsErrorLog: React.FC<TaskStatusLogProps> = ({ taskRun }) => {
+  const logDetails = getLogSnippet(taskRun);
```

Comment:
`getLogSnippet` only accepts `PipelineRun` instances and no `TaskRun` types here. The getLogSnippet method converts the given PipelineRun to a logDetails object which will be used by `LogSnippetBlock` to fetch the correct log.

For a PipelineRun this method return an object like this one:

```tsx
{
  containerName: "step-build",
  podName: "docker-bs1qu0-build-qmsfx-pod-5hpp4",
  title: "Failure on task build - check logs for details.",
}
```

But for the TaskRun it returns this message (extracted from yaml):

```tsx
{
  staticMessage: "\"step-build\" exited with code 1 (image: \"registry.redhat.io/rhel8/buildah@sha256:180c4d9849b6ab0e5465d30d4f3a77765cf0d852ca1cb1efb59d6e8c9f90d467\"); for logs run: kubectl -n christoph-test logs docker-bs1qu0-build-qmsfx-pod-5hpp4 -c step-build\n",
  title: "Failure - check logs for details.",
}
```

This is the reason why we see currently a different error log does on the PipelineRun Detail page and on the TaskRun detail page.

The PipelineRun detail page fetches the log from the Pod:

![image](https://user-images.githubusercontent.com/139310/96659023-45864d00-1346-11eb-9e2c-21ef1edaeb43.png)

But the TaskRun detail page shows an error message from the yaml:

![image](https://user-images.githubusercontent.com/139310/96659091-664ea280-1346-11eb-9726-46be726cdb2c.png)

I think we should write a similar getLogSnippet (or getLogDetails) for TaskRuns which also extracts the right podName and containerName (and shows the current message also as fallback).

The status steps contains all required infos for this:

```yaml
status:
  completionTime: '2020-10-21T00:24:58Z'
  conditions:
    - message: >
        "step-build" exited with code 1.... # FALLBACK
      reason: Failed
      status: 'False'
      type: Succeeded
  podName: docker-bs1qu0-build-qmsfx-pod-5hpp4
  steps:
    - container: step-generate
      name: generate
      terminated:
        exitCode: 0
        reason: Completed
    - container: step-build
      name: build
      terminated:
        exitCode: 1
        reason: Error    # I expect that we want see the log snippet of this container
```

Reply:
Could verify that this is solved now :+1: